python-login-server/server.py

```diff
@@ -25,30 +83,25 @@ def start(self):
       # Wait for a connection
       print('waiting for a connection')
       logging.debug('Waiting for a connection.')
-      connection, client_address = self.sock.accept()
+      connFd, client_address = self.sock.accept()
+      connection = Connection(connFd, client_address)
       try:
-        print('Connection from {address}'.format(address=client_address))
-        logging.info('Connection from {address}'.format(address=client_address))
-
         # Receive the data in small chunks and retransmit it
         while True:
-          data = connection.recv(16)
-          print('received {!r}'.format(data))
-          logging.info('Received {!r}.'.format(data))
-          if data:
-            print('sending data back to the client')
-            logging.debug('Sending data back to the client.')
-            connection.sendall(data)
-          else:
-            print('no data from {address}'.format(address=client_address))
-            logging.info('No data from {address}, closing connection.'.format(address=client_address))
-            break
+          connection.getMessage()
+          connection.handleMessage()
+      except Warning as warn:
+        logging.info(warn)
       finally:
-        # Clean up the connection
         connection.close()
 
+# client -> server: LOGIN <username|nem tartalmazhat spacet> <password>\n
```

Comment:
Nit: This should be moved elsewhere. It would be great to have it as a documentation.